fe/fe-core/src/main/java/com/starrocks/scheduler/mv/MVPCTRefreshListPartitioner.java

```diff
@@ -152,41 +158,98 @@ public Expr generatePartitionPredicate(Table refBaseTable, Set<String> refBaseTa
+    }
+
+    private static @Nullable Expr genPartitionPredicate(Table refBaseTable,
+                                                        Set<String> refBaseTablePartitionNames,
+                                                        List<Expr> refBaseTablePartitionSlotRefs,
+                                                        List<Column> refPartitionColumns,
+                                                        Map<String, PListCell> baseListPartitionMap) throws AnalysisException {
+        Preconditions.checkArgument(refBaseTablePartitionSlotRefs.size() == refPartitionColumns.size());
+        if (refPartitionColumns.size() == 1) {
+            boolean isContainsNullPartition = false;
+            List<Expr> sourceTablePartitionList = Lists.newArrayList();
+            Type partitionType = refPartitionColumns.get(0).getType();
+            for (String tablePartitionName : refBaseTablePartitionNames) {
+                PListCell cell = baseListPartitionMap.get(tablePartitionName);
+                for (List<String> values : cell.getPartitionItems()) {
+                    if (refPartitionColumns.size() != values.size()) {
+                        return null;
+                    }
+                    LiteralExpr partitionValue = new PartitionValue(values.get(0)).getValue(partitionType);
+                    if (partitionValue.isConstantNull()) {
+                        isContainsNullPartition = true;
+                        continue;
+                    }
+                    sourceTablePartitionList.add(partitionValue);
                 }
-                sourceTablePartitionList.add(partitionValue);
             }
-        }
-        Expr inPredicate = MvUtils.convertToInPredicate(mvPartitionSlotRef, sourceTablePartitionList);
-        // NOTE: If target partition values contain `null partition`, the generated predicate should
-        // contain `is null` predicate rather than `in (null) or = null` because the later one is not correct.
-        if (isContainsNullPartition) {
-            IsNullPredicate isNullPredicate = new IsNullPredicate(mvPartitionSlotRef, false);
-            return Expr.compoundOr(Lists.newArrayList(inPredicate, isNullPredicate));
+            Expr refBaseTablePartitionExpr = getRefBaseTablePartitionTransformExpr(refBaseTable,
+                    refPartitionColumns.get(0).getName(), (SlotRef) refBaseTablePartitionSlotRefs.get(0));
+            Expr inPredicate = MvUtils.convertToInPredicate(refBaseTablePartitionExpr, sourceTablePartitionList);
+            // NOTE: If target partition values contain `null partition`, the generated predicate should
+            // contain `is null` predicate rather than `in (null) or = null` because the later one is not correct.
+            if (isContainsNullPartition) {
+                IsNullPredicate isNullPredicate = new IsNullPredicate(refBaseTablePartitionExpr, false);
```

Comment:
if this IsNullPredicate is capable be eliminated, is it eliminated actually?

Reply:
I think it should be eliminated correctly, since generated partition predicates will be optimized again by Optimizer.